From 21e04d958622a9fd6fda1de311a47984b5dd1b7c Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Sat, 08 Feb 2020 20:48:59 +0200
Subject: [PATCH] [SW-8229] goya2: disable spi controller byte swap

Make sure at arcboot_rom has byte swapping disabled and
remove any other toggling in the code. It has to stay
disabled.

Change-Id: Iebf2e2d953863d4952364c98420e9d63554017b0
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/app/arcboot_rom.c b/app/arcboot_rom.c
index 46b092b..e377936 100644
--- a/app/arcboot_rom.c
+++ b/app/arcboot_rom.c
@@ -27,6 +27,9 @@
 {
 	uint32_t reg;
 
+	/* make sure spi controller doesn't swap bytes */
+	spi_flash_set_byte_swap(DISABLE_BYTE_SWAP);
+
 	/* on warm reset change default to reset ARC and PSOC */
 	reg = regmap_readl(mmPSOC_GLOBAL_CONF_PRSTN_RST_CFG_L);
 	reg |= PSOC_GLOBAL_CONF_PRSTN_RST_CFG_L_ARC_MASK;
diff --git a/common/dma.c b/common/dma.c
index 320482e..62311f9 100644
--- a/common/dma.c
+++ b/common/dma.c
@@ -18,9 +18,6 @@
 	uint16_t flash_offset_1k = flash_offset / DMA_SRC_BLOCK_SIZE;
 	uint16_t image_size_4b = (image_size / DMA_DST_BLOCK_SIZE);
 
-	/* TODO: check that in new RTL issue is fixed */
-	spi_flash_set_byte_swap(DISABLE_BYTE_SWAP);
-
 	regmap_writel(spi_dma_dst_addr_l,
 		      mmPSOC_GLOBAL_CONF_SPI_DMA_DST_ADDR_L);
 	regmap_writel(spi_dma_dst_addr_h,
diff --git a/common/spi_flash.c b/common/spi_flash.c
index 1f092e6..1176947 100644
--- a/common/spi_flash.c
+++ b/common/spi_flash.c
@@ -30,7 +30,6 @@
 
 uint32_t spi_flash_readl(uint32_t offset)
 {
-	spi_flash_set_byte_swap(ENABLE_BYTE_SWAP);
 	uint32_t reg = base_readl(HL_FLASH_BASE + offset);
 
 	return reg;
@@ -38,7 +37,6 @@
 
 void spi_flash_writel(uint32_t val, uint32_t offset)
 {
-	spi_flash_set_byte_swap(ENABLE_BYTE_SWAP);
 	base_writel(val, HL_FLASH_BASE + offset);
 }
 
